src/test/kotlin/org/rust/ide/intentions/AddImportForMatchArmPatternIntentionTest.kt

```diff
@@ -0,0 +1,129 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions
+
+class AddImportForMatchArmPatternIntentionTest : RsIntentionTestBase(AddImportForMatchArmPatternIntention::class) {
+
+    fun `test constant`() = doAvailableTest("""
+        mod inner {
+            pub const C: i32 = 0;
+        }
+
+        fn main() {
+            match 0 {
+                /*caret*/C => {}
+                _ => {}
+            }
+        }
+    """, """
+        use crate::inner::C;
+
+        mod inner {
+            pub const C: i32 = 0;
+        }
+
+        fn main() {
+            match 0 {
+                C => {}
+                _ => {}
+            }
+        }
+    """)
+
+    fun `test enum variant`() = doAvailableTest("""
+        enum E { A, B }
+        fn func(e: E) {
+            match e {
+                /*caret*/A => {}
+                B => {}
+            }
+        }
+    """, """
+        use crate::E::A;
+
+        enum E { A, B }
+        fn func(e: E) {
+            match e {
+                A => {}
+                B => {}
+            }
+        }
+    """)
+
+    fun `test complex pattern`() = doAvailableTest("""
+        mod inner {
+            pub const C: i32 = 0;
+        }
+
+        fn main() {
+            match (0, 0) {
+                (/*caret*/C, _) => {}
+            }
+        }
+    """, """
+        use crate::inner::C;
+
+        mod inner {
+            pub const C: i32 = 0;
+        }
+
+        fn main() {
+            match (0, 0) {
+                (C, _) => {}
+            }
+        }
+    """)
+
+    fun `test unavailable for ref pattern`() = doUnavailableTest("""
+        mod inner {
+            pub const C: i32 = 0;
+        }
+
+        fn main() {
+            match 0 {
+                ref /*caret*/C => {}
+                _ => {}
+            }
+        }
+    """)
+
+    fun `test unavailable for field binding`() = doUnavailableTest("""
+        mod inner {
+            pub const C: i32 = 0;
+        }
+
+        struct Foo { C: i32 }
+        fn func(foo: Foo) {
+            match foo {
+                Foo { /*caret*/C } => {}
```

Comment:
Actually, the intention can work in this case as well. Let's add a comment to explain current behaviour